tests_v2/test_browser_cli_agent_port.py

```
@@ -91,6 +91,32 @@ def _run_cli_text(cli: str, args: list[str], retries: int = 3) -> str:
+    for attempt in range(1, retries + 1):
+        proc = subprocess.run(
+            [cli, "--socket", SOCKET_PATH] + args,
+            capture_output=True,
+            text=True,
+            check=False,
+        )
+        if proc.returncode == 0:
+            try:
+                return json.loads(proc.stdout or "{}")
+            except Exception as exc:  # noqa: BLE001
+                raise cmuxError(f"Invalid CLI JSON output for {' '.join(args)}: {proc.stdout!r} ({exc})")
```

Ruff (0.15.4)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)